### ISSUE TITLE
Allow >100 limits when getting Admin Log Events Request (GetAdminLogRequest)

### ISSUE THREAD

**Issue by emma231** [LonamiWebs/Telethon#952]
From @Lonami's implements https://github.com/LonamiWebs/Telethon/commit/3d6c8915e3250d3ec1b26faa73cc08ad1f811081: get_message_history now allows return >100 message history.

Same have I encountered when I try to return >100 limits for GetAdminLogRequest. May I ask if this can be modified to allow >100 returns as well? 

The code is below. Specifically, I want to return 50000 results in GetAdminLogRequest.  
```
filter_results = ChannelAdminLogEventsFilter(
	join=True, 
	leave=False, 
	invite=False, 
	ban=False,
	unban=False, 
	kick=False, 
	unkick=False, 
	promote=False, 
	demote=False, 
	info=False,
	settings=False, 
	pinned=False, 
	edit=False, 
	delete=False
)
input_channel = InputChannel(channel_id=channel_id, access_hash = channel_access_hash)
log_request = GetAdminLogRequest(
	channel=input_channel,
	q = '',
	max_id=0,
	min_id=0,
	limit=50000,
	events_filter=filter_results
)

result = client(log_request)
```
If there's a way to work around this capped 100 limit without modifying Telethon client library, that works too.

Thanks!

**Comment by Lonami** [comment]
You can't do that, the request you're using is a *raw* request and that's not modified by the library. It would be necessary to make a convenience `client.get_admin_log`-like request.

**Comment by emma231** [comment]
Thanks for your fast reply, @Lonami! 

After reading your response, I found myself in serious help for 2 follow-up questions that made me further bewildered.

1. 

You said "It would be necessary to make a convenience client.get_admin_log-like request.". By this, do you mean: 

- **I need to write a function myself, (something called get_admin_log), in the TelegramClient class. Specifically, given that I cannot add it to your library, I need to first fork your repository, and then add get_admin_log to my own repository forked from yours.**

For this question, a 'yes' or 'no' maybe suffice. But for the next question, I think I get really trapped.

2. 

You said "The request(GetAdminLogRequest) you're using is a raw request." However:

- **If the new function 'get_admin_log' still invokes this raw request, how am I able to get >100 results?**

TO explain what I mean by this-- I only know that we can use a 'while' loop, and use an ``offset`` shifting window to capture all returns (>>100 limits), exactly like what you did to solve the limit issue in https://github.com/LonamiWebs/Telethon/commit/3d6c8915e3250d3ec1b26faa73cc08ad1f811081. However, unlike GetHistoryRequest, which takes ``offset`` as its parameter, [GetAdminLogRequest](https://lonamiwebs.github.io/Telethon/methods/channels/get_admin_log.html) has no ``offset`` parameter. Therefore, I think I am not able to modify ``offset`` in GetAdminLogRequest, even if I create a convenience function called ``client.get_admin_log``.

In summary, could you help me clarify in what way I can make >100 limits happen? Not because I tried to be lazy, but I am really stuck at my train of thought.

@Lonami Thank you!

**Comment by Lonami** [comment]
> given that I cannot add it to your library

You can. The project is open source so you can make a pull request. Furthermore, Python lets you subclass classes (e.g. `class MyClient(TelegramClient)`) to extend functionality. This is encouraged, but it would be good for everyone to benefit from some things.

> I need to first fork your repository, and then add get_admin_log to my own repository

Ah, yes, that's how it works here at GitHub. You can do this if you feel down for the task. You should probably take a look at how `iter_messages` is implemented, create `iter_admin_log`, and then create a `get_` variant for convenience.

> how am I able to get >100 results?

Offsets. You update the `max_id` to be the `min_id` of the last results so you can continue getting older ones. That's your offset.

**Comment by emma231** [comment]
That's incredibly helpful, and offers a couple of things I can try to see if I get a different result.

I will come back and update once I try a couple of things you suggested. Please leave this issue open, and I will update within a day or two. Thanks!

**Comment by emma231** [comment]
Hi dear @Lonami, 

> You should probably take a look at how iter_messages is implemented
> You update the max_id to be the min_id of the last results so you can continue getting older ones.

I have been following your suggestion, but I'm still not able to fetch joiner timestamps earlier than [last_id - 100](https://github.com/LonamiWebs/Telethon/blob/master/telethon/client/messages.py#L144).

After reading through [``iter_messages``](https://github.com/LonamiWebs/Telethon/blob/master/telethon/client/messages.py#L24), I think it's magic is to update ``offset_id`` in every call to gradually fetch all historical messages. 

However, 

- ``GetAdminLogRequest`` doesn't take ``offset_id`` as its parameter.

- Changing ``max`` and ``min_id`` within the GetAdminLogRequest doesn't give more results than 100. (Code pasted below).

- Other Telegram requests, like ``GetHistoryRequest ``, ``SearchRequest ``, simply doesn't return information about channel's joiner dates.

I want to get **all** historical timestamps about users who join in a telegram channel. With all these limitations, does it mean that my attempt to get full channel joiners' information is put at an dead end? Your help will be greatly appreciated. 

Thank you!

```
reverse_joiners = []
curr_max_id = 0
while True:
	tmp = []
	log_request = GetAdminLogRequest(
		channel=input_channel,
		q = '',
		max_id=curr_max_id,
		min_id=0,
		limit=100,
		events_filter=filter_results,
		admins=[]
	)
	result = client(log_request)
	if not result.events:
		break
	curr_max_id = result.events[-1].id
	for i in result.events:
		tmp.append(i.user_id)
	reverse_joiners.extend(tmp)
```

**Comment by Lonami** [comment]
Your code is right (although it should improve more), since it's [very similar to mine](https://gist.github.com/Lonami/02bb9d90d9b218e91ac8bb709927b58b#file-dump-admin-log-py-L27-L36) which does work.
